include/cpuinfo_riscv.h

```diff
@@ -0,0 +1,72 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CPU_FEATURES_INCLUDE_CPUINFO_RISCV_H_
+#define CPU_FEATURES_INCLUDE_CPUINFO_RISCV_H_
+
+#include "cpu_features_cache_info.h"
+#include "cpu_features_macros.h"
+
+#if !defined(CPU_FEATURES_ARCH_RISCV)
+#error "Including cpuinfo_riscv.h from a non-riscv target."
+#endif
+
+CPU_FEATURES_START_CPP_NAMESPACE
+
+typedef struct {
+    int riscv32 : 1;  // Is a 32 bit architecture
+    int riscv64 : 1;  // Is a 64 bit architecture
+    int riscv128 : 1; // Is a 128 bit architecture
+    int a : 1;        // Atomic Instructions
+    int c : 1;        // Compressed Instructions
+    int d : 1;        // Double Precision Floating Point
+    int e : 1;        // Base Integer Reduced
+    int f : 1;        // Single Precision Floating Point
+    int i : 1;        // Base Integer
+    int m : 1;        // Integer Multiplication and Division
+    int v : 1;        // Vector Operations
+    int q : 1;        // Quad Precision Floating Point
+} RiscvFeatures;
+
+
+typedef struct {
+    RiscvFeatures features;
+    char          uarch[64];  // 0 terminated string
+    char          vendor[64]; // 0 terminated string
```

Comment:
I think we should consider API that will be consistent with other operating systems, what about FreeBSD
and possible future operating systems? Can we get this info?

Reply:
I'm not sure about other os compatibility. The more general way would be using `misa` instruction but as we have seen in #247 it's not possible. Do you have any suggestion?

Reply:
I mean that we should define only common info, and avoid features which on certain operating systems will return empty results.

for example, if both Linux and FreeBSD return `hart` from the cpuinfo file, then we can add to our functionality, but if only one Linux/FreeBSD returns some info and the other doesn't provide that info we should avoid it.